Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceMain.mm

```diff
@@ -183,19 +184,21 @@ void XPCServiceEventHandler(xpc_connection_t peer)
             register_for_dlsym_callbacks();
 #endif
 
+            String serviceName = xpcDictionaryGetString(event, "service-name"_s);
+            if (!serviceName) {
+                RELEASE_LOG_ERROR(IPC, "XPCServiceEventHandler: 'service-name' is not present in the XPC dictionary");
+                return;
+            }
 #if PLATFORM(IOS_FAMILY)
             auto containerEnvironmentVariables = xpc_dictionary_get_value(event, "ContainerEnvironmentVariables");
             xpc_dictionary_apply(containerEnvironmentVariables, ^(const char *key, xpc_object_t value) {
                 setenv(key, xpc_string_get_string_ptr(value), 1);  // NOLINT
                 return true;
             });
+            if (!_set_user_dir_suffix(serviceName.latin1().data()))
```

Comment:
Can we log for both succeed and fail cases? Just to make it easy to notice change on temp directory is from log

Reply:
Will do! Thanks for reviewing!